tests/encode_images.rs

```diff
@@ -55,17 +55,17 @@ fn encode_decode() {
                         ifd::Value::UnsignedBig(8)
                     ])
                 ),
-                (Tag::Compression, ifd::Value::Unsigned(1)),
-                (Tag::PhotometricInterpretation, ifd::Value::Unsigned(2)),
+                (Tag::Compression, ifd::Value::Short(1)),
+                (Tag::PhotometricInterpretation, ifd::Value::Short(2)),
                 (Tag::StripOffsets, ifd::Value::Unsigned(8)),
-                (Tag::SamplesPerPixel, ifd::Value::Unsigned(3)),
+                (Tag::SamplesPerPixel, ifd::Value::Short(3)),
                 (Tag::RowsPerStrip, ifd::Value::Unsigned(3334)),
                 (Tag::StripByteCounts, ifd::Value::Unsigned(30000)),
                 (Tag::XResolution, ifd::Value::Rational(1, 1)),
                 (Tag::YResolution, ifd::Value::Rational(1, 1)),
-                (Tag::ResolutionUnit, ifd::Value::Unsigned(1)),
+                (Tag::ResolutionUnit, ifd::Value::Short(1)),
                 (Tag::Artist, ifd::Value::Ascii("Image-tiff".into())),
-                (Tag::Predictor, ifd::Value::Unsigned(1)),
+                (Tag::Predictor, ifd::Value::Short(1)),
```

Comment:
> The hard part is still [this concern](https://github.com/image-rs/image-tiff/issues/204#issuecomment-1566616975):
> 
> > This makes sense to fix, though the annoying part is figuring out whether anywhere else in the code is relying on the current behavior. Or whether it is likely to break any downstream users and thus might warrant additional caution

Looks like some tests added in #255 around November 2024 (after this PR was opened) discovered a potential area of breakage. Old behaviour is that some of these tags (e.g. Compression, PhotometricInterpretation, SamplesPerPixel, ResolutionUnit, Predictor) were incorrectly decoded as Unsigned (u32), whereas after this PR, they will be decoded as Short (u16).